nvflare/private/fed/client/process_aux_cmd.py

```diff
@@ -30,7 +30,8 @@ def process(self, req: Message, app_ctx) -> Message:
 
         shareable = pickle.loads(req.body)
 
-        result = engine.send_aux_command(shareable)
+        run_number = req.get_header(RequestHeader.RUN_NUM)
+        result = engine.send_aux_command(shareable, run_number)
```

Comment:
By doing this, we assume the app_ctx is an object that has this "send_aux_command" method.

I wonder if there is a way that we can enforce this specification.
Things like app_ctx need to be an object of ClientEngineInternalSpec

Reply:
We have this assumption in a lot of places, we don't need to fix it in this PR.

But I do think it would be better to find a way to specify / fix that.

Reply:
will address this with a separate PR. app_ctx in the RequestProcessor is a ClientEngine. Need to change all the RequestProcessor